src/Renci.SshNet/SshCommand.cs

```diff
@@ -294,6 +294,9 @@ public string EndExecute(IAsyncResult asyncResult)
                             _channel.Close();
                         }
 
+                        this._session.Disconnected -= Session_Disconnected;
+                        this._session.ErrorOccured -= Session_ErrorOccured;

```

Comment:
Good point. I'll find a better location. My use case did not have re-exeuction where the leak was produced, nor was I aware you could re-execute SshCommands.